From b3f4a9a64be4b6bcc81f13cf9677a486b42dd2b4 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Fri, 19 Mar 2010 17:23:12 -0700
Subject: [PATCH 02/14] XQuartz: Minor cleanup

Move RandRInit to where it will need to be (not yet implemented)

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
(cherry picked from commit 7da4f9df9c6490e3e604c170415d5af2d8c8ff29)
---
 hw/xquartz/quartz.c |   14 +++++++-------
 1 files changed, 7 insertions(+), 7 deletions(-)

diff --git xorg-server-1.7.6/hw/xquartz/quartz.c xorg-server-1.7.6/hw/xquartz/quartz.c
index c4142a8..3c04205 100644
--- xorg-server-1.7.6/hw/xquartz/quartz.c
+++ xorg-server-1.7.6/hw/xquartz/quartz.c
@@ -166,6 +166,11 @@ void QuartzInitOutput(
         FatalError("Could not register block and wakeup handlers.");
     }
 
+#if defined(RANDR) && !defined(FAKE_RANDR)
+    if(!QuartzRandRInit(pScreen))
+        FatalError("Failed to init RandR extension.\n");
+#endif
+
     // Do display mode specific initialization
     quartzProcs->DisplayInit();
 }
@@ -259,16 +264,11 @@ void QuartzUpdateScreens(void) {
     pScreen->width = width;
     pScreen->height = height;
     
-#ifndef FAKE_RANDR
-    if(!QuartzRandRInit(pScreen))
-        FatalError("Failed to init RandR extension.\n");
-#endif
-    
     DarwinAdjustScreenOrigins(&screenInfo);
     quartzProcs->UpdateScreen(pScreen);
     
-    sx = dixScreenOrigins[pScreen->myNum].x + darwinMainScreenX;
-    sy = dixScreenOrigins[pScreen->myNum].y + darwinMainScreenY;
+    sx = x + darwinMainScreenX;
+    sy = y + darwinMainScreenY;
     
     /* Adjust the root window. */
     pRoot = WindowTable[pScreen->myNum];
-- 
1.7.0.2

